gravitee-apim-console-webui/src/portal/customization/catalog/category/category.component.ts

```diff
@@ -0,0 +1,328 @@
+import { MatSelect } from '@angular/material/select';
+import { MatSlideToggle } from '@angular/material/slide-toggle';
+import { MatTooltip } from '@angular/material/tooltip';
+import { CdkDrag, CdkDragDrop, CdkDropList } from '@angular/cdk/drag-drop';
+
+import { GioPermissionModule } from '../../../../shared/components/gio-permission/gio-permission.module';
+import { GioGoBackButtonModule } from '../../../../shared/components/gio-go-back-button/gio-go-back-button.module';
+import {
+  GioApiSelectDialogComponent,
+  GioApiSelectDialogData,
+  GioApiSelectDialogResult,
+} from '../../../../shared/components/gio-api-select-dialog/gio-api-select-dialog.component';
+import { NewCategory } from '../../../../entities/category/NewCategory';
+import { UpdateCategory } from '../../../../entities/category/UpdateCategory';
+import { GioPermissionService } from '../../../../shared/components/gio-permission/gio-permission.service';
+import { SnackBarService } from '../../../../services-ngx/snack-bar.service';
+import { ApiV2Service } from '../../../../services-ngx/api-v2.service';
+import { CategoryV2Service } from '../../../../services-ngx/category-v2.service';
+import { CategoryService } from '../../../../services-ngx/category.service';
+import { Category } from '../../../../entities/category/Category';
+import { PortalHeaderComponent } from '../../../components/header/portal-header.component';
+
+interface ApiVM {
+  id: string;
+  name: string;
+  version: string;
+  contextPath: string;
+  order: number;
+}
+
+@Component({
+  selector: 'category',
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    FormsModule,
+    GioAvatarModule,
+    GioFormFilePickerModule,
+    GioFormSlideToggleModule,
+    GioGoBackButtonModule,
+    GioPermissionModule,
+    GioSaveBarModule,
+    MatButton,
+    MatCard,
+    MatCardContent,
+    MatCell,
+    MatCellDef,
+    MatColumnDef,
+    MatFormField,
+    MatHeaderCell,
+    MatHeaderRow,
+    MatHeaderRowDef,
+    MatHint,
+    MatIcon,
+    MatTableModule,
+    MatInput,
+    MatLabel,
+    MatOption,
+    MatRow,
+    MatRowDef,
+    MatSelect,
+    MatSlideToggle,
+    MatTable,
+    MatTooltip,
+    ReactiveFormsModule,
+    RouterLink,
+    PortalHeaderComponent,
+    MatDialogContent,
+    CdkDropList,
+    CdkDrag,
+  ],
+  templateUrl: './category.component.html',
+  styleUrl: './category.component.scss',
+})
+export class CategoryCatalogComponent implements OnInit {
+  mode: 'new' | 'edit' = 'new';
+
+  categoryDetails: FormGroup<{
+    name: FormControl<string>;
+    description: FormControl<string>;
+    highlightApi: FormControl<string>;
+  }>;
+  highlightApiControl: FormControl<string>;
+  categoryDetailsInitialValue: unknown;
+
+  category$: Observable<Category>;
+  apis$: Observable<ApiVM[]>;
+
+  displayedColumns = ['order', 'name', 'version', 'contextPath', 'actions'];
+
+  private refreshData = new BehaviorSubject(1);
+  private category = new BehaviorSubject<Category>(null);
```

Comment:
What is this BehaviorSubject being used for?

Reply:
It holds the current value of the category

Reply:
For what purpose? Where is it used in the component?